tests/appsec/waf/test_addresses.py

```diff
@@ -97,16 +97,20 @@ def test_specific_key(self):
     def setup_specific_key2(self):
         self.r_sk_4 = weblog.get("/waf/", headers={"X_Filename": "routing.yml"})
 
-    @missing_feature(library="python")
     @irrelevant(library="ruby", reason="Rack transforms underscores into dashes")
     @irrelevant(library="php", reason="PHP normalizes into dashes; additionally, matching on keys is not supported")
     @missing_feature(weblog_variant="spring-boot-3-native", reason="GraalVM. Tracing support only")
     def test_specific_key2(self):
         """attacks on specific header X_Filename, and report it"""
-
-        interfaces.library.assert_waf_attack(
-            self.r_sk_4, pattern="routing.yml", address="server.request.headers.no_cookies", key_path=["x_filename"]
-        )
+        try:
+            interfaces.library.assert_waf_attack(
+                self.r_sk_4, pattern="routing.yml", address="server.request.headers.no_cookies", key_path=["x_filename"]
+            )
+        except ValueError:
+            # also accept report on x-filename
```

Comment:
Name Python and WSGi here? (if that's the case) That will make it easier to later reason about where the exception comes from.

Reply:
Ruby seems to be impacted too, and php as well. I'm not sure this is really language specific, any new web framework could decide to do it as well.